partner_contact_in_several_companies/models/res_partner.py

```diff
@@ -129,26 +133,26 @@ def update_contact(self, vals):
         if contact_vals:
             self.with_context(__update_contact_lock=True).write(contact_vals)
 
-    @api.model
-    def _fields_sync(self, partner, update_values):
+    @api.multi
+    def _fields_sync(self, update_values):
         """Sync commercial fields and address fields from company and to
         children, contact fields from contact and to attached contact
         after create/update, just as if those were all modeled as
         fields.related to the parent
         """
-        super(ResPartner, self)._fields_sync(partner, update_values)
-        contact_fields = self._contact_fields()
-        # 1. From UPSTREAM: sync from parent contact
-        if update_values.get('contact_id'):
-            partner._contact_sync_from_parent()
-        # 2. To DOWNSTREAM: sync contact fields to parent or related
-        elif any(field in contact_fields for field in update_values):
-            update_ids = [
-                c.id for c in partner.other_contact_ids if not c.is_company
-            ]
-            if partner.contact_id:
-                update_ids.append(partner.contact_id.id)
-            self.browse(update_ids).update_contact(update_values)
+        super(ResPartner, self)._fields_sync(update_values)
+        for record in self:
```

Comment:
The [super method if not record looping](https://github.com/odoo/odoo/blob/5dc1c576011d801bf5f6bbb977cdd5373d178b54/odoo/addons/base/res/res_partner.py#L428), but instead operating on the recordset as if it were a singleton (without verifying). We should probably model that same pattern here and not perform the record loop